internal/output/html.go

```diff
@@ -0,0 +1,505 @@
+	}
+
+	// Split vulnerabilities into called and uncalled.
+	// Only add one vulnerability per group
+	ecosystemPrefix := models.Ecosystem(strings.Split(ecosystemName, ":")[0])
+	packageResults := processPackageResults(allVulns, groupIDs, uncalledVulnIDs, ecosystemPrefix)
+	var count HTMLVulnCount
+	for index, packageResult := range packageResults {
+		packageResults[index].Ecosystem = ecosystemName
+		packageResults[index].Source = packageSource.Source.Path
+		updateCount(&count, &packageResult.HTMLVulnCount)
+	}
+
+	return &HTMLSourceResult{
+		Source:               packageSource.Source.String(),
+		Ecosystem:            ecosystemName,
+		PackageResults:       packageResults,
+		CalledPackageCount:   len(calledPackages),
+		UncalledPackageCount: len(uncalledPackages),
+		HTMLVulnCount:        count,
+	}
+}
+
+// processPackageResults splits the given vulnerabilities into called and uncalled
+// based on the uncalledVulnIDs map.
+func processPackageResults(allVulns []HTMLVulnResult, groupIDs map[string]models.GroupInfo, uncalledVulnIDs map[string]bool, ecosystemPrefix models.Ecosystem) []HTMLPackageResult {
+	packageResults := make(map[string]*HTMLPackageResult)
+	for _, vuln := range allVulns {
+		groupInfo, isIndex := groupIDs[vuln.Summary.ID]
+		if !isIndex {
+			// We only display one vulnerability from one group
+			continue
+		}
+
+		// Add group IDs info
+		if len(groupInfo.IDs) > 1 {
+			vuln.Detail.GroupIDs = groupInfo.IDs[1:]
+		}
+
+		packageName := vuln.Summary.PackageName
+		packageResult, exist := packageResults[packageName]
+		if !exist {
+			packageResult = &HTMLPackageResult{
+				Name: packageName,
+			}
+			packageResults[packageName] = packageResult
+		}
+
+		// Get the max severity from groupInfo and increase the count
+		vuln.Summary.SeverityScore = groupInfo.MaxSeverity
+		vuln.Summary.SeverityRating, _ = severity.CalculateRating(vuln.Summary.SeverityScore)
+
+		if _, isUncalled := uncalledVulnIDs[vuln.Summary.ID]; isUncalled {
+			packageResult.UncalledVulns = append(packageResult.UncalledVulns, vuln)
+			packageResult.HTMLVulnCount.Uncalled = len(packageResult.UncalledVulns)
+
+			continue
+		}
+
+		packageResult.CalledVulns = append(packageResult.CalledVulns, vuln)
+		packageResult.HTMLVulnCount.Called = len(packageResult.CalledVulns)
+		addCount(&packageResult.HTMLVulnCount, vuln.Summary.SeverityRating)
+		if vuln.Summary.FixedVersion == UnfixedDescription {
+			packageResult.HTMLVulnCount.UnFixed += 1
+		} else {
+			packageResult.HTMLVulnCount.Fixed += 1
+		}
+	}
+
+	results := make([]HTMLPackageResult, 0, len(packageResults))
+	for _, result := range packageResults {
+		if len(result.CalledVulns) > 0 {
+			result.InstalledVersion = result.CalledVulns[0].Summary.InstalledVersion
+			result.FixedVersion = getMaxFixedVersion(ecosystemPrefix, result.CalledVulns)
+		}
+
+		results = append(results, *result)
+	}
+
+	// Sort packageResults to ensure consistent output
+	sort.Slice(results, func(i, j int) bool {
```

Comment:
Hmm does this also need to sort on version? E.g. if a package is repeated with 2 different versions installed.

Reply:
Good catch, updated